### PR TITLE
Added support for several functions provided by the v2 API

### PR DESCRIPTION
**Pull request by sidheart**
* No deletions or changes to existing code, only additions

### CODE REVIEW

File: signalfx/rest.py [hunk 1]
```diff
@@ -427,3 +468,124 @@ def delete_detector(self, detector_id, **kwargs):
...
+                    'offset': offset,
+                    'limit': batch_size,
+                    'name': name,
+                    'tags': tags or []
+                },
+                **kwargs)
+            resp.raise_for_status()
+            data = resp.json()
+            charts += data['results']
+            if len(charts) == data['count']:
```
Comment (mpetazzoni) [inline signalfx/rest.py]: This is dangerous if a chart is deleted while this loop pages through charts as `len(charts)` and `data['count']` may never be equal. You should use `>=` instead.

File: signalfx/rest.py [hunk 2]
```diff
@@ -427,3 +468,124 @@ def delete_detector(self, detector_id, **kwargs):
...
+    def get_chart(self, chart_id, **kwargs):
+        """Retrieve the single (v2) chart matching the given id.
+
+        Returns:
+            A (v2) chart as a JSON object (usable like a dictionary).
+        """
+        return self._get_object_by_id(self._CHART_ENDPOINT_SUFFIX, chart_id)
+
+    def get_charts(self, name=None, tags=None, batch_size=128, **kwargs):
+        """Retrieve all (v2) charts matching the given name; all (v2) charts up to batch_size otherwise.
```
Comment (mpetazzoni) [inline signalfx/rest.py]: It's not "up to batch_size". It'll be all v2 charts, paging through them in pages of batch_size elements.

File: signalfx/rest.py [hunk 3]
```diff
@@ -427,3 +468,124 @@ def delete_detector(self, detector_id, **kwargs):
...
+        """
+        return self._create_object(self._CHART_ENDPOINT_SUFFIX, chart)
+
+    # Functionality related to dashboards
+    def get_dashboard(self, dashboard_id, **kwargs):
+        """Retrieve the single (v2) chart matching the given id."""
+        return self._get_object_by_id(self._DASHBOARD_ENDPOINT_SUFFIX, dashboard_id)
+
+    def get_dashboards(self, name=None, batch_size=128, **kwargs):
+        """Retrieve all (v2) dashboards matching the given name; all (v2) dashboards up to batch_size otherwise.
```
Comment (mpetazzoni) [inline signalfx/rest.py]: Same about batch_size

File: signalfx/rest.py [hunk 4]
```diff
@@ -427,3 +468,124 @@ def delete_detector(self, detector_id, **kwargs):
...
+                params={
+                    'offset': offset,
+                    'limit': batch_size,
+                    'name': name
+                },
+                **kwargs)
+            resp.raise_for_status()
+            data = resp.json()
+            dashboards += data['results']
+            if len(dashboards) == data['count']:
```
Comment (mpetazzoni) [inline signalfx/rest.py]: Same about `>=`

File: signalfx/rest.py [hunk 5]
```diff
@@ -427,3 +468,124 @@ def delete_detector(self, detector_id, **kwargs):
...
+
+        Returns:
+            dictionary of the response (created dashboard model).
+        """
+        return self._create_object(self._DASHBOARD_ENDPOINT_SUFFIX, dashboard)
+
+    # Functionality related to dashboard groups
+    def get_dashboard_groups(self, name=None, batch_size=128, **kwargs):
+        """Retrieve all (v2) dashboard groups matching the given name;
+           all (v2) dashboard groups up to batch_size otherwise.
```
Comment (mpetazzoni) [inline signalfx/rest.py]: Same about batch_size

File: signalfx/rest.py [hunk 6]
```diff
@@ -427,3 +468,124 @@ def delete_detector(self, detector_id, **kwargs):
...
+                params={
+                    'offset': offset,
+                    'limit': batch_size,
+                    'name': name
+                },
+                **kwargs)
+            resp.raise_for_status()
+            data = resp.json()
+            dashboard_groups += data['results']
+            if len(dashboard_groups) == data['count']:
```
Comment (mpetazzoni) [inline signalfx/rest.py]: Same about `>=`

**Comment by cory-signalfx** [comment]
Closing due to inactivity. Some of this functionality has been added since, as well.

**Pull request closed without merging**